lib/ruby_lsp/ruby_lsp_rails/server.rb

```diff
@@ -380,17 +380,21 @@ def resolve_association_target(params)
         association_klass = const.reflect_on_association(params[:association_name].intern).klass
         source_location = Object.const_source_location(association_klass.to_s)
 
+        raise "Could not determine source location" unless source_location
+
         { location: source_location.first + ":" + source_location.second.to_s }
       rescue NameError
         nil
       end
 
+      #: (const: Module?) -> bool
       def active_record_model?(const)
         !!(
           const &&
             defined?(ActiveRecord) &&
             const.is_a?(Class) &&
             ActiveRecord::Base > const && # We do this 'backwards' in case the class overwrites `<`
+            const < ActiveRecord::Base && # this is just to narrow the type for Sorbet
```

Comment:
The reason why we used the inverted check `Class > const` is exactly to avoid using this one because we discovered through our work in Tapioca that some gems override `self.<`.

We should not include this check as this may return `false` even in cases where `const` does inherit from the expected class, if we hit one of those cases where it is overridden.

The correct solutions in this case are to either:

1. Fix Sorbet to understand that `Class > const` is the same as `const < Class`
2. Cast `const` as an `ActiveRecord::Base` after the line above

Reply:
Up to you. We can close or leave it open until what we need is supported.